bin.src/inspect_job.py

```diff
@@ -0,0 +1,121 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Print the contents of a persisted Job object.
+
+This script takes as arguments one or more Job .json files, and prints the
+top-level metadata and a summary of any measurements. It does not print
+metrics that don't have measurements (there are far too many) or
+specifications (which are not helpful for testing measurement code).
+"""
+
+import sys
+import json
+
+from lsst.verify import Job
+
+
+def is_measurement_metadata(key, metrics):
+    """Test whether a job-level metadata key is really measurement metadata.
+
+    Parameters
+    ----------
+    key : `str`
+        The key to test.
+    metrics : iterable of `lsst.verify.Name` or of `str`
+        The metrics recorded in the job.
+
+    Returns
+    -------
+    result : `bool`
+        `True` if ``key`` represents measurement metadata, `False` otherwise.
+    """
+    for metric in metrics:
+        if str(metric) in key:
+            return True
+    return False
+
+
+def simplify_key(key, prefix):
+    """Remove a prefix from a key, if it's present.
+
+    Parameters
+    ----------
+    key : `str`
+        The key to simplify.
+    prefix : `str`
+        The prefix to remove from ``key``.
+
+    Returns
+    -------
+    simplifiedKey : `str`
+        ``key`` with any initial ``prefix`` removed
+    """
+    if key.startswith(prefix):
+        return key.replace(prefix, "", 1)
+    else:
+        return key
+
+
+def inspect_job(filename):
+    """Present the measurements in a Job file.
+
+    The measurements and any metadata are printed to standard output.
+
+    Parameters
+    ----------
+    filename : `str`
+        The Job file to open. Must be in JSON format.
+    """
+    with open(filename) as f:
+        job = Job.deserialize(**json.load(f))
+
+    maxMetricLength = max([len(str(metric)) for metric in job.measurements])
+
+    print("Common metadata:")
+    for key, value in job.meta.items():
+        if is_measurement_metadata(key, job.measurements.keys()):
+            continue
+        print("%*s = %s" % (maxMetricLength, key, value))
+
+    print("\nMeasurements:")
+    for metric, measurement in job.measurements.items():
+        prettyQuantity = measurement.quantity.round(4)
+        if measurement.notes:
+            prefix = str(measurement.metric_name) + "."
+            simpleNotes = {simplify_key(key, prefix): value
+                           for key, value in measurement.notes.items()}
+            print("%*s = %10s (%s)"
+                  % (maxMetricLength, metric, prettyQuantity, simpleNotes))
+        else:
+            print("%*s = %10s" % (maxMetricLength, metric, prettyQuantity))
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print("Syntax: %s <job file> [[job file]...]" % sys.argv[0])
+        sys.exit(1)
+    for filename in sys.argv[1:]:
+        if len(sys.argv) > 2:
```

Comment:
Maybe better off putting this logic in `main()` and using `ArgumentParser` to manage the `argv` stuff.

Reply:
`ArgumentParser` seems kind of overkill here, don't you think? It's not a proper CLI, just a list of files.

Reply:
Either way. I find using argparse to even handle "just a list of files" to be more robust, but its your call.